doc/api/permissions.md

```diff
@@ -440,7 +446,152 @@ not adopt the origin of the `blob:` URL.
+
+Allowing access to spawning a process and creating worker threads can be done
+using the `--allow-child-process` and `--allow-worker` respectively.
+
+#### Runtime API
+
+When enabling the Permission Model through the [`--experimental-permission`][]
+flag a new property `permission` is added to the `process` object.
+This property contains two functions:
+
+* `permission.deny(scope [,parameters])`
+
+API call to deny permissions at runtime ([`permission.deny()`][])
+
+```js
+process.permission.deny('fs'); // Deny permissions to ALL fs operations
+
+// Deny permissions to ALL FileSystemWrite operations
+process.permission.deny('fs.write');
+// deny FileSystemWrite permissions to the protected-folder
+process.permission.deny('fs.write', ['/home/rafaelgss/protected-folder']);
+// Deny permissions to ALL FileSystemRead operations
+process.permission.deny('fs.read');
+// deny FileSystemRead permissions to the protected-folder
+process.permission.deny('fs.read', ['/home/rafaelgss/protected-folder']);
+```
+
+API call to check permissions at runtime ([`permission.has()`][])
+
+```js
+process.permission.has('fs.write'); // true
+process.permission.has('fs.write', '/home/rafaelgss/protected-folder'); // true
+
+process.permission.deny('fs.write', '/home/rafaelgss/protected-folder');
+
+process.permission.has('fs.write'); // true
+process.permission.has('fs.write', '/home/rafaelgss/protected-folder'); // false
+```
+
+#### File System Permissions
+
+To allow access to the filesystem, use the flag `--allow-fs`:
+
+```console
+$ node --experimental-permission --allow-fs=read index.js
+Hello world!
+(node:19836) ExperimentalWarning: Permission is an experimental feature
+(Use `node --trace-warnings ...` to show where the warning was created)
+```
+
+The valid arguments for the `--allow-fs` flag option are:
+
+* `write` - To manage `FileSystemWrite` (writing) operations.
+* `read` - To manage `FileSystemRead` (reading) operations.
+* `fs` - To allow all the `FileSystem` operations.
+
+Example:
+
+* `--allow-fs=read:/tmp/` - It will allow `FileSystemRead` access to the `/tmp/`
+  folder.
+* `--allow-fs=read:/tmp/:/home/.gitignore` - It allows `FileSystemRead` access
+  to the `/tmp/` folder **and** the `/home/.gitignore` path.
+
+You can also mix both arguments:
+
+* `--allow-fs=write,read:/tmp/` - It will allow `FileSystemRead` access to the
+  `/tmp/` folder **and** allow **all** the `FileSystemWrite` operations.
+* **Note**: It accepts wildcard parameters as well.
+  For instance: `--allow-fs=write:/home/test*` will allow everything that
+  matches the wildcard. e.g: `/home/test/file1` or `/home/test2`
+
+There are constraints you need to know before using this system:
+
+* Native modules are not restricted by the Permission Model.
+* Relative paths are not supported through the CLI (`--allow-fs`).
+  The runtime API supports relative paths.
+* Permission changes are not retroactively applied to existing resources.
+  Consider the following snippet:
+
```

Comment:
can we also document here the inheritance behaviour of the permission model?
 - the model does / does not inherit to a child node process
 - the model does / nodes not inherit to a worker thread

Reply:
Sure! Good point